### PR TITLE
Append the required fields of embedded structs

### PR DESCRIPTION
**Pull request by XieX**
Appends the required fields of embedded structs so they don't get missed.

e.g. these are equivalent:

```
type Bird struct {
	name string `required:"true"`
	AirspeedVelocity float32 `json:"airspeedVelocity"`
}
```

```
type Bird struct {
	Animal
	AirspeedVelocity float32 `json:"airspeedVelocity"`
}

type Animal struct {
	name string `required:"true"`
}
```

But in the latter case the output for the `Bird` schema would not be marking `name` as required.

### CODE REVIEW

File: parser.go [hunk 1]
```diff
@@ -1587,6 +1587,7 @@ func (p *parser) parseAstField(pkgPath, pkgName string, structSchema *SchemaObje
 					}
 					structSchema.Properties.Set(propertyName, refPropertySchema)
 				}
+				structSchema.Required = append(structSchema.Required, refSchema.Required...)
```
Comment (XieX) [inline parser.go:1590]: This is the only functional change. Everything else is test/fixture updates :P

File: example/example.json [hunk 2]
```diff
@@ -20,6 +20,24 @@
...
+                  "$ref": "#/components/schemas/Bird"
+                }
+              }
+            }
+          }
+        },
+        "summary": "Get all birds",
+        "operationId": "getBirds"
+      }
+    },
```
Comment (XieX) [inline example/example.json:40]: This and the two new schema for `Animal` and `Bird` should be the only meaningful changes to this spec, but when I regenerated it seemed to make some unrelated changes - apologies for that.

**Review by jazanne — APPROVED** [review]
Love that this was such a small change

Confirmed that it fixed the audit log issues we were seeing 
Screen Shot 2021-08-06 at 11 06 37 AM

File: parser.go (hunk 1 above)
Comment (InTheCloudDan) [reply parser.go:1590]: And there's no way `refSchema` could be empty? Or it doesn't matter anyway if it is as it just won't have and `required` properties to unpack?

Comment (XieX) [reply parser.go:1590]: Yeah, we're still within the `if ok {` block, which I believe is guarding against `refSchema` not having been found in `p.KnownIDSchema`.

Comment (XieX) [reply parser.go:1590]: I guess it's possible that `nil` was inserted into `KnownIDSchema` for the given id, and we'll get a panic if that's the case - we probably want that though because that would be a bug.
